enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -1090,13 +1125,33 @@ func (s *ExecutionServer) cacheExecuteResponse(ctx context.Context, taskID strin
 	return cachetools.UploadActionResult(ctx, s.env.GetActionCacheClient(), arn, ar)
 }
 
+func (s *ExecutionServer) cacheActionResult(ctx context.Context, actionResourceName *digest.ResourceName, response *repb.ExecuteResponse, md *espb.ExecutionAuxiliaryMetadata) error {
+	if response.GetCachedResult() || md.GetExecutionTask() == nil {
+		return nil
+	}
+	// The action executed and the executor sent us an ExecutionTask in the
+	// auxiliary metadata, which indicates we need to write to the cache.
+	action := md.GetExecutionTask().GetAction()
+	// If the action failed or do_not_cache is set, upload information about the error via a failed
+	// ActionResult under an invocation-specific digest, which will not ever be seen by bazel but
+	// may be viewed via the Buildbuddy UI.
```

Comment:
I forgot to mention this earlier - we don't need to store failed ActionResults in the cache anymore, because the UI relies on the ExecuteResponse stored with `cacheExecuteResponse` to render the results now (both in the success and failure cases)